src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -1019,28 +1013,37 @@ var ReactDOMFiberComponent = {
           if (expectedStyle !== serverValue) {
             warnForPropDifference(propKey, serverValue, expectedStyle);
           }
-        } else if (
-          isCustomComponentTag ||
-          DOMProperty.isCustomAttribute(propKey)
-        ) {
+        } else if (isCustomComponentTag) {
           // $FlowFixMe - Should be inferred as not undefined.
-          extraAttributeNames.delete(propKey);
+          extraAttributeNames.delete(propKey.toLowerCase());
```

Comment:
Is it not obvious to me why this is necessary. Do we have a test verifying this change? Can you add a comment as to why we do this?